Mathlib.lean

```diff
@@ -2252,6 +2252,7 @@ import Mathlib.Data.String.Lemmas
 import Mathlib.Data.Subtype
 import Mathlib.Data.Sum.Basic
 import Mathlib.Data.Sum.Interval
+import Mathlib.Data.Sum.LE
```

Comment:
Maybe `Mathlib.Data.Sum.Order`? Or should this go under `Mathlib.Order` directly? CC @YaelDillies

Reply:
I think it belongs to the `Data` folder, but I don't have any opinion about naming this particular file.

Reply:
Yeah for now this should all go `Data.Sum.Order`, and later it will be moved to `Order.Sum` or something.

Reply:
Are you sure about the former file? Its docstring says:
```
# Orders on a sum type

This file defines the disjoint sum and the linear (aka lexicographic) sum of two orders and
provides relation instances for `Sum.LiftRel` and `Sum.Lex`.
```

Reply:
I mean genuinely your lemmas are so basic that they could go in `Order.Basic`

Reply:
I thought `Data` was more basic than folders dedicated to respective topics. Was I wrong?

Reply:
Also, do we want LT versions? I noticed the lemmas in the mentioned file had both LE and LT versions.

Reply:
I just tried `Order.Basic` but in this file the instance that would synthesize `LE (β → α)` from `[LE α]` is missing.

Reply:
Alright, since the previous suggestion didn't work without increasing imports in the basic file, I moved it all to `Algebra.Order.Sum`.